### ISSUE TITLE
Error handling: Invalid UTF-8 in input fasta not wrapped

### ISSUE THREAD

**Issue by corneliusroemer** [nextstrain/nextclade#978]
By accident I put a zip folder as input fasta and noticed that Nextclade did not report the file in which the error had occurred - the error seemed unhandled.

See screenshot:
<img width="995" alt="image" src="https://user-images.githubusercontent.com/25161793/187728045-ce2191eb-00ea-4446-9599-22e85a3429d9.png">

I think it would be nice to bubble up and say something like: in file X, problem Y

**Comment by ivan-aksamentov** [comment]
Yeah, looks like it crashed parser in an ugly way ant the whole thread along with it.
Error handing in presence of scoped threads is a bit tricky, but I need to do it sometimes... Maybe also migrate from `crossbeam` to the shiny new [`std::scope`](https://blog.rust-lang.org/2022/08/11/Rust-1.63.0.html#scoped-threads) from Rust 1.63.
